### PR TITLE
Added webpack prod config

### PR DESCRIPTION
**Pull request by CarloPalinckx**
### This PR:

**Adds** ✨
- The `uglify-webpack-plugin`

**Changes** 🌀
- Dist artifacts to be minified and without sourcemaps
- Moves `@types/react` and `@types/react-dom` to dependecies so that they will be installed with the prod env.
- Circle configuration

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,26 +1,40 @@
...
     working_directory: ~/repo
+
+version: 2
+
+jobs:
+  setup:
+    <<: *defaults
     steps:
+      - attach_workspace:
+          at: .
```
Comment (woutsluiter) [inline .circleci/config.yml]: Indenting seems a bit off in this file (2 and 4 spaces)

Comment (CarloPalinckx) [reply .circleci/config.yml]: the 4 spaces here are required by yaml :)

Comment (woutsluiter) [reply .circleci/config.yml]: Could be wrong here, but it looks like there are still 2 flavors used in this file. L2 has four spaces, while on L11 2 spaces are used.

Comment (CarloPalinckx) [reply .circleci/config.yml]: Ah I think you are right here, yaml need double spaces after a `-` property but not up there. Good spot 👍

**Pull request merged**